core/lib/vm_interface/src/types/outputs/execution_result.rs

```diff
@@ -103,3 +105,50 @@ impl VmExecutionResultAndLogs {
         }
     }
 }
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum TxExecutionStatus {
+    Success,
+    Failure,
+}
+
+impl TxExecutionStatus {
+    pub fn from_has_failed(has_failed: bool) -> Self {
+        if has_failed {
+            Self::Failure
+        } else {
+            Self::Success
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct TransactionExecutionResult {
+    pub transaction: Transaction,
+    pub hash: H256,
+    pub execution_info: VmExecutionMetrics,
+    pub execution_status: TxExecutionStatus,
+    pub refunded_gas: u64,
+    pub operator_suggested_refund: u64,
+    pub compressed_bytecodes: Vec<CompressedBytecodeInfo>,
+    pub call_traces: Vec<Call>,
```

Comment:
Now that I've poked around it a bit, I think `Call` can be moved to `vm_interface` as well. Will try to do separately to keep this PR reasonably sized.